packages/zikaron/src/zikaron/dashboard/search.py

```
@@ -89,145 +89,141 @@ def search(self, query: str, n_results: int = 10) -> List[Tuple[int, float]]:
-                        doc_id = self.ids[i]
-                        break
-                
-                if doc_id:
-                    rrf_scores[doc_id] += (1 - alpha) / (k + rank + 1)
-            
+                # Use metadata to find ID
+                doc_id = meta.get("source_file", "") + ":" + str(rank)
+                rrf_scores[doc_id] += (1 - alpha) / (k + rank + 1)
+
             # 4. Sort by combined RRF score and return top results
             sorted_results = sorted(rrf_scores.items(), key=lambda x: x[1], reverse=True)[:n_results]
```

Ruff (0.14.14)

[error] Local variable `sorted_results` is assigned to but never used

Remove assignment to unused variable `sorted_results`

(F841)